### ISSUE TITLE
State of the project

### ISSUE THREAD

**Issue by samuelcolvin** [pydantic/FastUI#368]
TL;DR this project is on hold while the Pydantic team is flat out building [logfire](https://pydantic.dev/logfire) and continuing to maintain Pydantic. 

We use FastUI internally for the admin interface of Logfire, and it works well for that application, but we don't have a strong impetus to invest heavily in improving FastUI.

---

Some background: I originally built FastUI in the hope we could use it for some user facing functionality in Logfire, that didn't happen for a bunch of reasons:
1. FastUI wasn't ready when we needed to build that stuff
2. There isn't enough repetitive CRUD functionality in Logfire to demand the truely reusable/composable component structure that FastUI tries to provide.
3. There was some complex use of generic unions we needed in Pydantic before arbitrary components would work in FastUI, which slowed us down.

Ultimately I couldn't persuade our frontend team that FastUI was the right tool to use, and I wasn't prepared to force them to use a tool I was still experimenting with.

Also, as per #48 I would really like the rendering of HTML in FastUI to happen exclusively (or mostly) serverside, but that's a big rewrite that I don't have time to work on right now.

I won't mark this repo as archived so discussions can continue, and we can continue to fix any bugs we find. But don't expect new PRs or issues to receive attention from the Pydantic team.

For those who were excited about FastUI, sorry to disappoint. 🙏 

Maybe at some point in future I'll have the time and bandwidth to resume the project.

**Comment by fmrib00** [comment]
Such a pity! but thanks very much for your hard work on such a useful package, which I have been using heavily and successfully in my real-life project!

**Comment by Zaubeerer** [comment]
Thanks for the honesty and clarity!

Maybe there is a group of developers that would like to pick up the work?

**Comment by dsommer** [comment]
@samuelcolvin why not put this notice in the main README so it shows up on the main page. I started working on a project using fastUI before I knew it was on hold. Seems like a great project but would be nice to know it is on hold on the main website and not in an issue...

**Comment by samuelcolvin** [comment]
Good idea, pr welcome. Otherwise will do soon.

**Comment by samuelcolvin** [comment]
notice added to readme.

**Comment by dsommer** [comment]
Thank you, sorry I have been meaning to do a PR.

@samuelcolvin I also wanted to say thank you for all you tremendous work on Pydantic, Pydantic AI and logfire. I have used Pydantic in multiple projects now and it is great addition to the python ecosystem. I have just started using Pydantic AI! 
Thank You

**Comment by samuelcolvin** [comment]
no problem, thanks so much for trying this things!

**Comment by nibsbin** [comment]
I noticed some recent activity preparing for version 0.8.0. Is there a future for FastUI? I'm super excited about a frontend-backend divorced UI framework in Python.

Also, echoing @dsommer 's  sentiment: yall are amazing! I'm looking forward to expressing ideas fast with pydantic.ai and hopefully FastUI.
